be/src/agent/task_worker_pool.cpp

```diff
@@ -169,6 +167,18 @@ void TaskWorkerPool::start() {
 #endif
 }
 
+void TaskWorkerPool::stop() {
+    if (_stopped) {
+        return;
+    }
+    _stopped = true;
+    std::lock_guard l(_worker_thread_lock);
+    _worker_thread_condition_variable->notify_all();
+    for (uint32_t i = 0; i < _worker_count; ++i) {
+        _worker_threads[i].join();
```

Comment:
I think you need to release the mutex first before joining the threads, otherwise, the threads cannot be waked up.